game-app/game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/retreat/OffensiveSubsRetreatTest.java

```diff
@@ -107,21 +103,17 @@ void hasNameWhenDestroyerIsOnDefenseAndWithdrawIsAfterBattle() {
   }
 
   @Test
-  void hasNoNamesWhenDestroyerIsOnDefenseAndWithdrawIsBeforeBattle() {
+  void hasNameWhenDestroyerIsOnDefenseAndWithdrawIsBeforeBattle() {
     final BattleState battleState =
         givenBattleStateBuilder()
             .attackingUnits(List.of(givenUnitCanEvade()))
             .defendingUnits(List.of(givenUnitDestroyer()))
-            .gameData(
-                MockGameData.givenGameData()
-                    .withSubmersibleSubs(true)
-                    .withSubRetreatBeforeBattle(true)
-                    .build())
+            .gameData(MockGameData.givenGameData().withSubmersibleSubs(true).build())
             .build();
     final OffensiveSubsRetreat offensiveSubsRetreat =
         new OffensiveSubsRetreat(battleState, battleActions);
 
-    assertThat(offensiveSubsRetreat.getNames(), is(empty()));
+    assertThat(offensiveSubsRetreat.getNames(), hasSize(1));
```

Comment:
nit, mentioning in the assertThat comment what we expect in the 'offensiveSubsRetreat' would be good context. If this were to fail in the future due to the collection being empty, the coder/maintainer/bug-fixer might wonder why this test expected the collection to be non-emtpy.

Reply:
Fixed in https://github.com/triplea-game/triplea/pull/9098/commits/977c2cef1f80931235b9b0a68459f1298df945d3

Reply:
:+1: